### PR TITLE
[OUDS] Docs: Add 'Getting started > Accessibility' page

### PR DESCRIPTION
**Pull request by louismaximepiton**
### Related issues

Listed in https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/2589

### Description

This PR adds the "Getting Started > Accessibility" page based on:
- the previous [corresponding Boosted page](https://boosted.orange.com/docs/5.3/getting-started/accessibility/) (see [code source](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/site/content/docs/5.3/getting-started/accessibility.md))
- the [corresponding Bootstrap page](https://getbootstrap.com/docs/5.3/getting-started/accessibility/) (see [code source](https://github.com/twbs/bootstrap/blob/main/site/content/docs/5.3/getting-started/accessibility.md))

Please that not all content can be reintroduced in this first PR, as some components or concepts are not yet described in the rest of the documentation.

There are some Bootstrap inconsistencies with WCAG in the doc.

- [ ] Modals component
- [ ] Dropdowns component
- [ ] Tooltips component
- [ ] Color utility
- [ ] Background utility
- [ ] Carousels component
- [ ] Spinners component
- [ ] Focus mixin and design defined
- [ ] Scroll target style
- [ ] Target size mixin
- [ ] Text utility

### Types of change

- New documentation (non-breaking change which adds functionality)

### Live previews

- https://deploy-preview-2675--boosted.netlify.app/docs/0.0/getting-started/accessibility/

### CODE REVIEW

File: site/content/docs/0.0/getting-started/accessibility.md [hunk 1]
```diff
@@ -8,4 +8,100 @@ aliases:
 toc: true
 ---
 
-{{< callout-soon "page" >}}
+OUDS Web provides an easy-to-use framework of ready-made styles, layout tools, and interactive components, allowing developers to create websites and applications that are visually appealing, functionally rich, and accessible out of the box.
+
+## Overview and limitations
+
+The overall accessibility of any project built with OUDS Web depends in large part on the author's markup, additional styling, and scripting they've included. However, provided that these have been implemented correctly, it should be perfectly possible to create websites and applications with OUDS Web that fulfill [<abbr title="Web Content Accessibility Guidelines">WCAG</abbr> 2.2](https://www.w3.org/TR/WCAG/) (A/AA/AAA), [Section 508](https://www.section508.gov/), and similar accessibility standards and requirements.
```
Comment (julien-deramond) [inline site/content/docs/0.0/getting-started/accessibility.md:15]: WCAG 2.1 → WCAG 2.2 modification should be done in Bootstrap first. And globally if necessary.

File: site/content/docs/0.0/getting-started/accessibility.md [hunk 2]
```diff
@@ -8,4 +8,100 @@ aliases:
...
+
+### Structural markup
+
+OUDS Web's styling and layout can be applied to a wide range of markup structures. This documentation aims to provide developers with best practice examples to demonstrate the use of OUDS Web itself and illustrate appropriate semantic markup, including ways in which potential accessibility concerns can be addressed.
+
+### Interactive components
+
+OUDS Web's interactive components<!--—such as modal dialogs, dropdown menus, and custom tooltips—--> are designed to work for touch, mouse, and keyboard users. Through the use of relevant [<abbr title="Web Accessibility Initiative">WAI</abbr>-<abbr title="Accessible Rich Internet Applications">ARIA</abbr>](https://www.w3.org/WAI/standards-guidelines/aria/) roles and attributes, these components should also be understandable and operable using assistive technologies (such as screen readers).
```
Comment (julien-deramond) [inline site/content/docs/0.0/getting-started/accessibility.md:23]:
```suggestion
OUDS Web's interactive components<!--—such as modal dialogs, dropdown menus, and custom tooltips—-->are designed to work for touch, mouse, and keyboard users. Through the use of relevant [<abbr title="Web Accessibility Initiative">WAI</abbr>-<abbr title="Accessible Rich Internet Applications">ARIA</abbr>](https://www.w3.org/WAI/standards-guidelines/aria/) roles and attributes, these components should also be understandable and operable using assistive technologies (such as screen readers).
```

File: site/content/docs/0.0/getting-started/accessibility.md [hunk 3]
```diff
@@ -8,4 +8,100 @@ aliases:
...
+
+Because OUDS Web's components are purposely designed to be fairly generic, authors may need to include further <abbr title="Accessible Rich Internet Applications">ARIA</abbr> roles and attributes, as well as JavaScript behavior, to more accurately convey the precise nature and functionality of their component. This is usually noted in the documentation.
+
+<!--### Color contrast
+
+Some combinations of colors that currently make up OUDS Web's default palette—used throughout the framework for things such as button variations, alert variations, form validation indicators—may lead to *insufficient* color contrast (below the recommended [WCAG 2.2 text color contrast ratio of 4.5:1](https://www.w3.org/TR/WCAG/#contrast-minimum) and the [WCAG 2.2 non-text color contrast ratio of 3:1](https://www.w3.org/TR/WCAG22/#non-text-contrast)), particularly when used against a light background.
```
Comment (julien-deramond) [inline site/content/docs/0.0/getting-started/accessibility.md:29]: We should use the same link as Bootstrap.

```suggestion
Some combinations of colors that currently make up OUDS Web's default palette—used throughout the framework for things such as button variations, alert variations, form validation indicators—may lead to *insufficient* color contrast (below the recommended [WCAG 2.1 text color contrast ratio of 4.5:1](https://www.w3.org/TR/WCAG/#contrast-minimum) and the [WCAG 2.1 non-text color contrast ratio of 3:1](https://www.w3.org/TR/WCAG/#non-text-contrast)), particularly when used against a light background.
```

File: site/content/docs/0.0/getting-started/accessibility.md [hunk 4]
```diff
@@ -8,4 +8,100 @@ aliases:
...
+
+Unlike Bootstrap, in OUDS Web **contrasts are locked to ensure they meet [WCAG 2.2 accessibility standards for color contrast](https://www.w3.org/TR/WCAG22/#contrast-minimum)** when using [`.text-*`]({{< docsref "/utilities/colors" >}}) and [`.bg-*` utilities]({{< docsref "/utilities/background" >}}), by defining `color` and `background-color` altogether. Please refer to [our theme colors]({{< docsref "/customize/color-palette" >}}) to have a full preview of OUDS Web color palette’s reached WCAG level.-->
```
Comment (julien-deramond) [inline site/content/docs/0.0/getting-started/accessibility.md:31]:
```suggestion
Unlike Bootstrap, in OUDS Web **contrasts are locked to ensure they meet [WCAG 2.1 accessibility standards for color contrast](https://www.w3.org/TR/WCAG/#contrast-minimum)** when using [`.text-*`]({{< docsref "/utilities/colors" >}}) and [`.bg-*` utilities]({{< docsref "/utilities/background" >}}), by defining `color` and `background-color` altogether. Please refer to [our theme colors]({{< docsref "/customize/color-palette" >}}) to have a full preview of OUDS Web color palette’s reached WCAG level.-->
```

File: site/content/docs/0.0/getting-started/accessibility.md [hunk 5]
```diff
@@ -8,4 +8,100 @@ aliases:
...
+
+### Maximum line length
+
+When writing a paragraph, it is commonly admitted that a line should have 80 characters as a maximum. This phenomenon is carefully explained in the [C20 technique: Using relative measurements to set column widths](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/C20).
+
+<!--For more information, please [read text utilities]({{< docsref "/utilities/text#line-length" >}}).-->
+
+## Additional resources
+
+- [Web Content Accessibility Guidelines (WCAG) 2.2](https://www.w3.org/TR/WCAG/)
```
Comment (julien-deramond) [inline site/content/docs/0.0/getting-started/accessibility.md:97]:
```suggestion
- [Web Content Accessibility Guidelines (WCAG) 2.1](https://www.w3.org/TR/WCAG/)
```

**Review by julien-deramond — CHANGES_REQUESTED** [review]

File: site/content/docs/0.0/getting-started/accessibility.md (hunk 1 above)
Comment (louismaximepiton) [reply site/content/docs/0.0/getting-started/accessibility.md:15]: Yep, I'll do it Bs side first, I'll revert it in here.

File: site/content/docs/0.0/getting-started/accessibility.md [hunk 6]
```diff
@@ -8,4 +8,100 @@ aliases:
...
+<!--#### Under a fixed header
+
+When using a fixed (or sticky) header, tabbing backward often hides focused element under the header. OUDS Web sets `scroll-padding-top` property for such case. This feature is configurable in two ways:
+
+1. `$scroll-offset-top` variable defines the offset,
+2. and [`$enable-fixed-header` allows to drop this rule]({{< docsref "/customize/options" >}}) if you don't use a fixed header.
+
+{{< scss-docs name="scroll-offset" file="scss/_reboot.scss" >}}-->
+
+<!--### Minimum target size
```
Comment (julien-deramond) [inline site/content/docs/0.0/getting-started/accessibility.md:83]: This one could probably be uncommented right away, isn't it?

File: site/content/docs/0.0/getting-started/accessibility.md [hunk 7]
```diff
@@ -8,4 +8,100 @@ aliases:
...
+
+On browsers that support `prefers-reduced-motion`, and where the user has *not* explicitly signaled that they'd prefer reduced motion (i.e. where `prefers-reduced-motion: no-preference`), OUDS Web enables smooth scrolling using the `scroll-behavior` property.
+
+### Focus visibility
+
+OUDS Web includes [WICG's `:focus-visible` polyfill](https://github.com/WICG/focus-visible) to ensure an enhanced focus visibility for keyboard users while shutting down focus styles on active state.
+
+{{< scss-docs name="focus-visibility" file="scss/_reboot.scss" >}}
+
+<!--OUDS Web provides `focus-visible()` mixin to ensure a proper visible focus state:
```
Comment (julien-deramond) [inline site/content/docs/0.0/getting-started/accessibility.md:62]: The lines from here to 68 could be uncommented right away, isn't it?

**Review by julien-deramond — APPROVED** [review]

**Pull request merged**